prototype_source/maskedtensor_overview.rst

```diff
@@ -0,0 +1,334 @@
+    tensor([[0.3548,    nan, 0.0000],
+            [0.6452,    nan, 1.0000],
+            [0.0000,    nan, 0.0000]])
+
+:class:`MaskedTensor` result:
+
+    >>> mt.softmax(0)
+    MaskedTensor(
+      [
+        [  0.3548,       --,       --],
+        [  0.6452,       --,   1.0000],
+        [      --,       --,       --]
+      ]
+    )
+
+`Issue 61474 -- Implementing missing torch.nan* operators <https://github.com/pytorch/pytorch/issues/61474>`__
+--------------------------------------------------------------------------------------------------------------
+
+In the above issue, there is a request to add additional operators to cover the various `torch.nan*` applications,
+such as ``torch.nanmax``, ``torch.nanmin``, etc.
+
+In general, these problems lend themselves more naturally to masked semantics, so instead of introducing additional
+operators, we propose using :class:`MaskedTensor`s instead. Since
+`nanmean has already landed <https://github.com/pytorch/pytorch/issues/21987>`__, we can use it as a comparison point:
+
+    >>> x = torch.arange(16).float()
+    >>> y = x * x.fmod(4)
+    >>> z = y.masked_fill(y == 0, float('nan'))  # we want to get the mean of y when ignoring the zeros
+    >>> y
+    tensor([ 0.,  1.,  4.,  9.,  0.,  5., 12., 21.,  0.,  9., 20., 33.,  0., 13.,
+            28., 45.])
+    >>> z   # z is just y with the zeros replaced with nan's
+    tensor([nan,  1.,  4.,  9., nan,  5., 12., 21., nan,  9., 20., 33., nan, 13.,
+            28., 45.])
+    >>> y.mean()
+    tensor(12.5000)
+    >>> z.nanmean()
+    tensor(16.6667)
+    >>> torch.mean(masked_tensor(y, y != 0))  # MaskedTensor successfully ignores the 0's
+    MaskedTensor( 16.6667, True)
+
+In the above example, we've constructed a `y` and would like to calculate the mean of the series while ignoring
+the zeros. `torch.nanmean` can be used to do this, but we don't have implementations for the rest of the
+`torch.nan*` operations. :class:`MaskedTensor` solves this issue by being able to use the base operation,
+and we already have support for the other operations listed in the issue. For example:
+
+    >>> torch.argmin(masked_tensor(y, y != 0))
+    MaskedTensor(  1.0000, True)
+
+Indeed, the index of the minimum argument when ignoring the 0's is the 1 in index 1.
+
+:class:`MaskedTensor` can also support reductions when the data is fully masked out, which is equivalent
+to the case above when the data Tensor is completely ``nan``. ``nanmean`` would return ``nan``
+(an ambiguous return value), while MaskedTensor would more accurately indicate a masked out result.
+
+    >>> x = torch.empty(16).fill_(float('nan'))
+    >>> x
+    tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan])
+    >>> torch.nanmean(x)
+    tensor(nan)
+    >>> torch.mean(masked_tensor(x, ~torch.isnan(x)))
+    MaskedTensor(--, False)
+
+This is a similar problem to safe softmax where `0/0 = nan` when what we really want is an undefined value.
```

Comment:
Add Further Reading 

Further Reading
----------------
* Link to the next MaskedTensor Tutorial